landing_page/spec/controllers/user_controller_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+    context "create a new user account with acceptable credentials" do
+
+        it "successfully adds a new user" do
+            expect{create_new_user}.to change(User, :count).by(1)
+        end
+
+        it "logs the user in upon account creation" do
+            create_new_user
+            expect(session[:user_id]).to_not be_nil
+        end
+
+        it "redirects a new user to their profile page upon creation" do
+            create_new_user
+            expect(response).to redirect_to(assigns(:user))
+        end
+
+    end
+
+    context "GET #index" do
+
+        it "shows the list of users, expects a redirect response to be 200" do
```

Comment:
Ideally, test scenarios should not have expectations as part of their names. A scenario should describe what should happen, "show the list of users", but not how that verification is made. The point here is that the method of verification could change and you would have to update the scenario name just because of it.

And in this specific case, what is the evidence that the `#index` shows the list of the users? Just a `200` response? Is that enough? If I have 10 users and my list only shows 5, but returns a `200` status is that correct?

Reply:
I have added further tests for the index. The 200 response isn't adequate on it's own.

Please see commit fa0508cf969ee02dc88928e9c412d18df6ea8a74 for updated tests